airflow/www/security.py

```diff
@@ -457,25 +457,25 @@ def add_homepage_access_to_custom_roles(self):
 
         self.get_session.commit()
 
-    def get_all_permissions(self):
+    def get_all_permissions(self) -> Set[Tuple[str, str]]:
         """Returns all permissions as a set of tuples with the perm name and view menu name"""
-        perms = set()
-        for permission_view in self.get_session.query(self.permissionview_model).all():
-            if permission_view.permission and permission_view.view_menu:
-                perms.add((permission_view.permission.name, permission_view.view_menu.name))
-
-        return perms
+        return set(
+            self.get_session.query(self.permissionview_model)
+            .join(self.permission_model)
+            .join(self.viewmenu_model)
+            .with_entities(self.permission_model.name, self.viewmenu_model.name)
+            .all()
+        )
```

Comment:
Name isn't nullable on both permission and viewmodel, and with inner joins we are guaranteed to get both which lets us return directly instead of walking the whole list to check them.